src/MerchantCenter/PriceBenchmarks.php

```diff
@@ -43,18 +234,36 @@ public function update_price_benchmarks(): void {
 			$query->reload_data();
 
 			// Insert new benchmark data.
-			foreach ( $benchmarks as $benchmark ) {
+			foreach ( $benchmarks as $id => $benchmark ) {
+				$performance_data = $this->get_performance_data( $id );
```

Comment:
Now that we're getting all the Metrics data at once, let's avoid individual API calls based on IDs and instead query all the `MerchantPerformanceView` data in a single query and map it to the `PriceCompetitivenessProductView` and `PriceInsightsProductView` in one process.